src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -439,26 +452,17 @@ private void UpdateSortedValues(EnhancedSortedSet<TValue> bucket, TValue? previo
         /// <summary>
         /// Actual removal mechanism.
         /// </summary>
-        protected virtual bool Remove(TKey key, TValue value)
+        protected virtual bool Remove(TKey key, out TValue? value)
         {
-            // Always remove from worst values; as may have been where it came from and dangling item
-            bool removed = _worstSortedValues.Remove(value);
```

Comment:
this couldn't work because value comes from `_cacheMap.TryGetValue` in the first place